paddle/go/optimizer/optimizer_factory.h

```diff
@@ -0,0 +1,74 @@
+#ifndef PADDLE_OPTIMIZER_FACTORY_H_
+#define PADDLE_OPTIMIZER_FACTORY_H_
+
+#include "parameter_optimizer.h"
+
+namespace paddle {
+namespace optimizer {
+
+
+template <class T>
+class SGDOptimizer : public ParameterOptimizer<T> {
```

Comment:
This file contains declaration for `SGDOptimizer `, `AdagradOptimizer`... Maybe it's better to move them to different headers, like: `sgd_optimizer.h`, `adagrad_optimizer.h`.

Btw, I think factory typically mean one function that creates instances. Here is the declaration of different optimizers, so `optimizer_factory.h` is probably not a good name.